File: server/Models/Product.cs
namespace SfaApi.Models
{
	public class Product
	{
		public int Id { get; set; }

		// ── Required (LKAST) ──
		public string Name { get; set; } = null!;                // Item Description
		public string? ItemNo { get; set; }                      // Item No.
		public string? Quality { get; set; }                     // QUALITY
		public string Category { get; set; } = "Tiles";         // Series: Tiles, Marble, Granite, Sanitaryware, Other
		public string? Size { get; set; }                        // e.g. "600x600", "800x1200"
		public decimal? Weight { get; set; }                     // WT (weight per unit)
		public decimal? BoxCoverage { get; set; }                // Box Sqr. Mtr
		public decimal? KgPerBox { get; set; }                   // KG Per Box
		public string? Code { get; set; }                        // Double Name / Item Code
		public string? Remarks { get; set; }                     // Remarks

		// ── Optional ──
		public string? Description { get; set; }
		public string? ImageUrl { get; set; }                    // product image path/URL
		public string? Thickness { get; set; }                   // e.g. "9mm", "12mm"
		public string? Finish { get; set; }                      // Glossy, Matt, Rustic, Satin, Carving, High Gloss
		public string? Shade { get; set; }                       // Light, Medium, Dark
		public string? Type { get; set; }                        // Floor, Wall, Outdoor, Marble, Other
		public int? PiecesPerBox { get; set; }

		// ── Pricing ──
		public decimal? RatePerSqm { get; set; }                 // Rate per SQM (required)
		public decimal Price { get; set; }                       // MRP per unit
		public decimal? DealerPrice { get; set; }                // price for dealers
		public string Unit { get; set; } = "Box";               // Box, SqFt, Pcs

		// ── Tags ──
		public bool IsNewArrival { get; set; }
		public bool IsDiscontinued { get; set; }
		public bool IsActive { get; set; } = true;
        public bool IsArchived { get; set; } = false;   // soft-delete flag

		// ── Timestamps ──
		public DateTime CreatedAt { get; set; }
		public DateTime? UpdatedAt { get; set; }
	}
}

File: mobile/app/src/main/java/com/example/sfa/Product.kt
package com.example.sfa

data class Product(
    val id: Int = 0,
    val name: String = "",
    val description: String = "",
    val itemNo: String = "",               // Item No.
    val quality: String = "",              // Quality grade
    val code: String = "",
    val remarks: String = "",              // Remarks
    val imageUrl: String = "",
    val category: String = "Tiles",       // Series: Tiles, Marble, Granite, Sanitaryware, Other
    val size: String = "",                 // e.g. "600x600"
    val weight: Double? = null,            // WT (weight per unit)
    val thickness: String = "",
    val finish: String = "",               // Glossy, Matt, Rustic, etc.
    val shade: String = "",                // Light, Medium, Dark
    val type: String = "",                 // Floor, Wall, Outdoor, Marble, Other
    val boxCoverage: Double? = null,       // Box Sqr.Mtr per box
    val kgPerBox: Double? = null,           // KG per box
    val ratePerSqm: Double? = null,         // Rate per SQM
    val piecesPerBox: Int? = null,
    val price: Double = 0.0,               // MRP (fallback if ratePerSqm missing)
    val dealerPrice: Double? = null,
    val unit: String = "Box",
    val isNewArrival: Boolean = false,
    val isDiscontinued: Boolean = false,
    val isActive: Boolean = true
)

/** Mirrors /api/product-config response — all lists come from DB, fallback to built-in defaults. */
data class ProductConfig(
    val category: List<String> = listOf("Tiles", "Marble", "Granite", "Sanitaryware", "Other"),
    val size: List<String> = listOf("600x600", "800x800", "300x600", "300x300"),
    val quality: List<String> = listOf("Premium", "Standard", "Block"),
    val type: List<String> = listOf("Floor", "Wall", "Outdoor", "Marble", "Other"),
    val finish: List<String> = listOf("Glossy", "Matt", "Rustic", "Satin", "High Gloss", "Other"),
    val shade: List<String> = listOf("Light", "Medium", "Dark"),
    val unit: List<String> = listOf("Box", "SqFt", "Pcs")
) {
    companion object {
        val Default = ProductConfig()
    }
}
